charts/opentelemetry-collector/templates/_config.tpl

```diff
@@ -131,22 +131,15 @@ service:
 receivers:
   filelog:
     include: [ /var/log/pods/*/*/*.log ]
+    {{- if not .Values.agentCollector.containerLogs.includeAgentLogs }}
     # Exclude collector container's logs. The file format is /var/log/pods/<namespace_name>_<pod_name>_<pod_uid>/<container_name>/<run_id>.log
     exclude: [ /var/log/pods/{{ .Release.Namespace }}_{{ include "opentelemetry-collector.fullname" . }}*_*/{{ .Chart.Name }}/*.log ]
+    {{- end }}
     start_at: beginning
     include_file_path: true
     include_file_name: false
     operators:
-      # Find out which format is used by kubernetes
-      - type: router
-        id: get-format
-        routes:
-          - output: parser-docker
-            expr: '$$record matches "^\\{"'
-          - output: parser-crio
-            expr: '$$record matches "^[^ Z]+ "'
-          - output: parser-containerd
-            expr: '$$record matches "^[^ Z]+Z"'
+      {{- if eq .Values.agentCollector.containerLogs.containerRunTime "cri-o" }}
```

Comment:
I do not fully understand what this change does. Did this change delete the ability to automatically detect the format used by kubernetes and requires the user to specify the format?

Reply:
It looks like that

Reply:
@rockb1017 please clarify why want to change this. It appears to me automatic detection is valuable since there is one less thing for users to worry about.

Reply:
yes, instead of collector detecting the format, we are allowing users to provide their container runtime. this is how it is done in SCK. Though I am not saying this is definitely better way, but at least it will definitely be better for performance. I would say asking users to specify the container runtime once and have more efficient pipeline is better.

Reply:
It is a tradeoff between ease of use and performance. Both are valid approach and different users likely will have different preferences. I think what we can do is by default have automatic detection of the format and in addition to that allow the user to specify the format, in which case the detection will be disabled and we will have better performance.